src/map/swipe.js

```diff
@@ -0,0 +1,147 @@
+import angular from 'angular';
+import * as olEvents from 'ol/events.js';
+
+/**
+ * @type {angular.IModule}
+ * @hidden
+ */
+const module = angular.module('ngeoMapswipe', []);
+
+module.run(
+  /**
+   * @ngInject
+   * @param {angular.ITemplateCacheService} $templateCache
+   */
+  $templateCache => {
+    // @ts-ignore: webpack
+    $templateCache.put('ngeo/src/map/swipe', require('./swipe.html'));
+  }
+);
+
+module.value(
+  'ngeoMapswipeTemplateUrl',
+  /**
+   * @param {angular.IAttributes} $attrs Attributes.
+   * @return {string} The template url.
+   */
+  $attrs => {
+    const templateUrl = $attrs.ngeoMapswipeTemplateUrl;
+    return templateUrl !== undefined ? templateUrl : 'ngeo/src/map/swipe';
+  }
+);
+
+/**
+ * @param {angular.IAttributes} $attrs Attributes.
+ * @param {function(angular.IAttributes): string} ngeoMapswipeTemplateUrl Template function.
+ * @return {string} Template URL.
+ * @ngInject
+ * @private
+ * @hidden
+ */
+function ngeoMapswipeTemplateUrl($attrs, ngeoMapswipeTemplateUrl) {
+  return ngeoMapswipeTemplateUrl($attrs);
+}
+
+/**
+ * The controller for the Mapswipe component.
+ */
+class SwipeController {
+  /**
+   * @param {angular.IScope} $scope Angular scope.
+   * @param {JQuery} $element Element.
+   * @param {angular.auto.IInjectorService} $injector Main injector.
+   * @constructor
+   * @private
+   * @hidden
+   * @ngInject
+   * @ngdoc controller
+   * @ngname ngeoMapswipeController
+   */
+  constructor($scope, $element, $injector) {
+    /**
+     * @type {import('ol/Map.js').default}
+     */
+    this.map;
+
+    /**
+     * @type {import('ol/layer/Layer.js').default}
+     */
+    this.layer;
+
+    /**
+     * @type {angular.IScope}
+     * @private
+     */
+    this.scope_ = $scope;
+
+    /**
+     * @type {number}
+     */
+    this.swipeValue = 50;
+
+    /**
+     * @type {JQuery}
+     */
+    this.swipeInput_ = $element.find('.swipe-input');
+
+    /**
+     * @type {?import("ol/events.js").EventsKey}
+     * @private
+     */
+    this.layerKey_ = null;
+  }
+
+  /**
+   * Init the controller
+   */
+  $onInit() {
+    this.layerKey_ = this.layer.on('prerender', this.handleLayerPrerender_.bind(this));
```

Comment:
push the result of this in the listener key array

instead of .bind, add `this` as 3rd argument